lib/rules/sort-vars.js

```diff
@@ -16,7 +16,29 @@ module.exports = function(context) {
 
     return {
         "VariableDeclaration": function(node) {
-            node.declarations.reduce(function(memo, decl) {
+
+            // Creates new node declaration
+            var nodeDeclarations = node.declarations.reduce(function(memo, decl) {
+
+                // Checks for ObjectPattern and maps new declarations
+                if (decl.id.type === "ObjectPattern") {
```

Comment:
You also need an exception for ArrayPattern, or else you'll get the same error for destructed arrays.